### PR TITLE
finish question 6, refactor tests slightly

### PR DESCRIPTION
**Pull request by MattyB2112**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -24,141 +24,192 @@ describe("/api", () => {
...
+      .get("/api/articles/")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+        expect(articles[0].comment_count).toBe(2);
+        expect(articles[1].comment_count).toBe(1);
+        expect(articles[2].comment_count).toBe(0);
+        expect(articles[3].comment_count).toBe(0);
+        expect(articles[4].comment_count).toBe(0);
+        expect(articles[5].comment_count).toBe(2);
```
Comment (pippaamy) [inline __tests__/app.test.js:102]: I dont think this is necessary  - we could use a forEach and say it was a value of number :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -24,141 +24,192 @@ describe("/api", () => {
...
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        body.forEach((comment) => {
+          expect(comment.hasOwnProperty("comment_id")).toBe(true);
+          expect(comment.hasOwnProperty("votes")).toBe(true);
+          expect(comment.hasOwnProperty("created_at")).toBe(true);
+          expect(comment.hasOwnProperty("author")).toBe(true);
+          expect(comment.hasOwnProperty("body")).toBe(true);
+          expect(comment.hasOwnProperty("article_id")).toBe(true);
```
Comment (pippaamy) [inline __tests__/app.test.js:187]: we know the article id will be 1 :)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -24,141 +24,192 @@ describe("/api", () => {
...
+      .then(({ body }) => {
+        expect([body]).toBeSortedBy("created_at");
+      });
+  });
+  test("invalid article id sends 400 error and message", () => {
+    return request(app)
+      .get("/api/articles/banana/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("bad request");
```
Comment (pippaamy) [inline __tests__/app.test.js:204]: what if its a 200 but theres no comments yet?

**Review by pippaamy — COMMENTED** [review]
great job, some changes to think about :)

File: __tests__/app.test.js (hunk 3 above)
Comment (MattyB2112) [reply __tests__/app.test.js:204]: Hi Pippa! I've sorted my code taking into account your other comments but I'm struggling with this one...will there ever be a 200 and no comments? the endpoint is to find the comments for that article, so my thinking is that there wouldn't be a 200 and no comments as the endpoint is meant to take the user to the comments, so wouldn't it just be a 404...?

Comment (pippaamy) [reply __tests__/app.test.js:204]: if the article exists but doesnt have any comments itll be 200 but if like below the article doesnt exist thats a 404

**Pull request merged**